File: UseItApp.Domain/Models/Loan.cs
namespace UseItApp.Domain.Models;

public enum LoanStatus
{
    Requested,
    Approved,
    Rejected,
    Active,
    ReturnInitiated, 
    Returned,
    Overdue
}

public class Loan
{
    public int Id { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public DateTime? ActualReturnDate { get; set; }
    public LoanStatus Status { get; set; } = LoanStatus.Requested;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public int ItemId { get; set; }
    public int BorrowerId { get; set; }
    
    public Item Item { get; set; } = null!;
    public User Borrower { get; set; } = null!;
}

File: UseItApp.API/ClientApp/src/app/models/loan.ts
import { User } from './user';
import { Item } from './item';

export enum LoanStatus {
  Requested,
  Approved,
  Rejected,
  Active,
  ReturnInitiated,
  Returned,
  Overdue,
}

export interface Loan {
  id?: number;
  startDate: Date;
  endDate: Date;
  actualReturnDate?: Date;
  status: LoanStatus;
  notes?: string;
  createdAt?: Date;
  itemId: number;
  borrowerId: number;
  item?: Item;
  borrower?: User;
}
